File: java-schema-sdk/src/main/java/org/openmhealth/schema/domain/omh/AreaUnitValue.java
/*
 * Copyright 2015 Open mHealth
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.openmhealth.schema.domain.omh;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import org.openmhealth.schema.serializer.SerializationConstructor;

import java.math.BigDecimal;


/**
 * @author Emerson Farrugia
 * @version 1.0
 * @see
 * <a href="http://www.openmhealth.org/documentation/#/schema-docs/schema-library/schemas/omh_area-unit-value">area-unit-value</a>
 */
public class AreaUnitValue extends TypedUnitValue<AreaUnit> {

    public static final SchemaId SCHEMA_ID = new SchemaId(OMH_NAMESPACE, "area-unit-value", "1.0");

    @SerializationConstructor
    protected AreaUnitValue() {
    }

    @JsonCreator
    public AreaUnitValue(@JsonProperty("unit") AreaUnit unit, @JsonProperty("value") BigDecimal value) {
        super(unit, value);
    }

    @Override
    public SchemaId getSchemaId() {
        return SCHEMA_ID;
    }

    public AreaUnitValue(AreaUnit unit, double value) {
        super(unit, value);
    }

    public AreaUnitValue(AreaUnit unit, long value) {
        super(unit, value);
    }
}

File: kotlin-schema-sdk/src/main/kotlin/org/openmhealth/schema/domain/omh/AreaUnitValue.kt
package org.openmhealth.schema.domain.omh

import org.openmhealth.schema.domain.OMH_NAMESPACE
import org.openmhealth.schema.domain.SchemaId
import org.openmhealth.schema.domain.SchemaSupport
import java.math.BigDecimal

/**
 * An area.
 *
 * @author Emerson Farrugia
 * @version 1.0
 * @see [area-unit-value](http://www.openmhealth.org/documentation/#/schema-docs/schema-library/schemas/omh_area-unit-value)
 */
data class AreaUnitValue(
    val unit: AreaUnit,
    val value: BigDecimal
) : SchemaSupport {

    constructor(unit: AreaUnit, value: Double) : this(unit, BigDecimal.valueOf(value))

    override val schemaId: SchemaId
        get() = SchemaId(OMH_NAMESPACE.value, "area-unit-value", "1.0")
}
